src/Functions/h3GetRes0Indexes.cpp

```diff
@@ -0,0 +1,86 @@
+#include "config_functions.h"
+
+#if USE_H3
+
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
+#include <IO/WriteHelpers.h>
+#include <Common/typeid_cast.h>
+#include <base/range.h>
+
+#include <h3api.h>
+
+
+namespace DB
+{
+namespace
+{
+
+class FunctionH3GetRes0Indexes final : public IFunction
+{
+public:
+    static constexpr auto name = "h3GetRes0Indexes";
+
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionH3GetRes0Indexes>(); }
+
+    std::string getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 0; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeUInt64>());
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName &, const DataTypePtr & result_type, size_t input_rows_count) const override
+    {
+        if (input_rows_count == 0)
+            return result_type->createColumn();
+
+        auto data = ColumnArray::create(ColumnUInt64::create());
+        /// res0_indexes contains all indexes with resolution 0 as returned by the h3 getRes0Cells function
+        const Array res0_indexes
```

Comment:
Here, we just need to create ColumnArray, then create ColumnUInt64, and in ColumnUInt64 pass result of h3 function call.
Current implementation will always allocate big array of `Field` type, and writing such constants manual by hand are error-prone.

Reply:
I've changed to remove hard coded constant.
However, I need some help to figure out how to pass results of h3 func to ColumnUInt64 directly without copying the results into an intermediate Array and then inserting that into ColumnArray. When I tried to do `return ColumnArray::create(std::move(res))` but I'm unable to do it without the offsets.

Reply:
@kitaisreal can you take another look here. Would be nice to have some pointers to instantiate the result of h3 function call directly to ColumnArray without having to deal with offsets or intermediate array.